### PR TITLE
support Op MaxPooWthArgmax in CPU and WebGL

### PR DESCRIPTION
**Pull request by pyu10055**
Adding support for MaxPooWthArgmax op https://www.tensorflow.org/api_docs/python/tf/nn/max_pool_with_argmax

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3010)

### CODE REVIEW

File: tfjs-core/src/backends/webgl/kernels/MaxPoolWithArgmax_impl.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import {TensorInfo} from '../../../kernel_registry';
+import * as conv_util from '../../../ops/conv_util';
+import {MathBackendWebGL} from '../backend_webgl';
+import {Pool2DProgram} from '../pool_gpu';
+
+export function maxPoolWithArgmaxImpl(
+    x: TensorInfo, includeBatchInIndex: boolean, convInfo: conv_util.Conv2DInfo,
+    backend: MathBackendWebGL): TensorInfo[] {
+  // if (convInfo.filterWidth === 1 && convInfo.filterHeight === 1 &&
```
Comment (lina128) [inline tfjs-core/src/backends/webgl/kernels/MaxPoolWithArgmax_impl.ts:26]: Remove?

File: tfjs-converter/python/tensorflowjs/op_list/convolution.json [hunk 2]
```diff
@@ -79,6 +79,50 @@
...
+        "name": "strides",
+        "type": "number[]"
+      },
+      {
+        "tfName": "padding",
+        "name": "pad",
+        "type": "string"
+      },
+      {
+        "tfName": "data_format",
```
[the remark this answers is not in the document]
Comment (pyu10055) [reply tfjs-converter/python/tensorflowjs/op_list/convolution.json:104]: good eyes, I have removed the data_format field, the other field is for output format, we do not used those.

File: tfjs-core/src/backends/webgl/pool_gpu.ts [hunk 3]
```diff
@@ -98,7 +99,14 @@ export class Pool2DProgram implements GPGPUProgram {
               if (value ${compareOp} currMinMaxValue) {
                 minMaxValue = value;
                 minMaxValueFound = 1.0;
-                minMaxPosition = wR * ${effectiveFilterWidth} + wC;
+                minMaxPosition = ${
+          flattenPositions ?
```
[the remark this answers is not in the document]
Comment (pyu10055) [reply tfjs-core/src/backends/webgl/pool_gpu.ts:103]: good suggestion, I simplify the statement, with compiled string.

File: tfjs-core/src/backends/webgl/kernels/MaxPoolWithArgmax_impl.ts (hunk 1 above)
Comment (pyu10055) [reply tfjs-core/src/backends/webgl/kernels/MaxPoolWithArgmax_impl.ts:26]: done

File: tfjs-core/src/backends/cpu/kernels/MaxPoolWithArgmax_impl.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import {computeStrides} from '../../../util';
+import {MathBackendCPU} from '../backend_cpu';
+
+export function maxPoolWithArgmaxImpl(
+    xValues: TypedArray, xShape: number[], dtype: DataType,
+    includeBatchInIndex: boolean, convInfo: Conv2DInfo,
+    backend: MathBackendCPU) {
+  const strides = computeStrides(xShape);
+  const maxPools =
+      backend.pool(xValues, xShape, dtype, strides, convInfo, 'max');
```
[the remark this answers is not in the document]
Comment (pyu10055) [reply tfjs-core/src/backends/cpu/kernels/MaxPoolWithArgmax_impl.ts:28]: good suggestion, moved functions to a new file.

**Pull request merged**